tools/roslaunch/src/roslaunch/config.py

```diff
@@ -448,11 +448,16 @@ def load_config_default(roslaunch_files, port, roslaunch_strs=None, loader=None,
     load_roscore(loader, config, verbose=verbose)
 
     # load the roslaunch_files into the config
-    for i, f in enumerate(roslaunch_files):
-        args = roslaunch_args[i] if roslaunch_args else None
+    for f in roslaunch_files:
+        args = roslaunch_args.get(f)
         try:
-            logger.info('loading config file %s'%f)
-            loader.load(f, config, argv=args, verbose=verbose)
+            if args and len(args) and hasattr(args[0],'__iter__'): # non-empty list of iterables
```

Comment:
Allow for one launchfile with multiple distinct calls by allowing the `args`  itself to be a list of arglists. The file will be launched once for each `arglist` in `args`. Otherwise, just launch the file with `args` as normal.